atomica/plotting.py

```diff
@@ -1606,36 +1613,53 @@ def _turn_off_border(ax) -> None:
     ax.yaxis.set_ticks_position('left')
 
 
-def plot_legend(entries: dict, plot_type='patch', fig=None):
+def plot_legend(entries: dict, plot_type='patch', fig=None, legendsettings=None):
     """
     Render a new legend
 
     :param entries: Dict where key is the label and value is the colour e.g. `{'sus':'blue','vac':'red'}`
-    :param plot_type: can be 'patch' or 'line'
+    :param plot_type: can be 'patch' or 'line', or a list the same length as param_entries
     :param fig: Optionally takes in the figure to render the legend in. If not provided, a new figure will be created
+    :param legendsettings: settings for the layout of the legend. If not provided will default to appropriate values depending on whether the legend is separate or together with a plot
     :return: The matplotlib `Figure` object containing the legend
 
     """
+    if plot_type is None:
```

Comment:
If you think that `line` is a more sensible default than `patch` then I think we could either change `plot_type` to default to `line` in the function declaration, or if the issue is that you want to actually pass in `None` as the `plot_type` from the calling code, have `plot_type` default to `None` the same was as `fig` does

Reply:
Ah yeah, I'm not sure why I did that, so either more logical solution is fine!

Also the whole overloading of "plot_type" and allowing it to be a list is really messy and could perhaps be straightened out, I just did that to be backwards compatible.